lib/backend/watchersyncer/watchercache.go

```diff
@@ -432,13 +410,10 @@ func (wc *watcherCache) markAsValid(resourceKey string) {
 	}
 }
 
-// onError signals to the syncer that this watcherCache is not in-sync if the number of consecutive errors
-// exceeds the error threshold.  See finishResync() for how the watcherCache goes back to in-sync.
+// onError signals to the syncer that this watcherCache is not in-sync, hence perform resync.
+// See finishResync() for how the watcherCache goes back to in-sync.
```

Comment:
Just spoke with @fasaxc .   Felix does not expect a transition to go from in-sync back to waiting-for-datastore.

I think we can simply remove onError().